lib_nbgl/src/nbgl_use_case.c

```diff
@@ -1402,15 +1392,33 @@ static uint8_t nbgl_useCaseGetNbPagesForContent(const nbgl_content_t *content, u
     nbElements = getContentNbElement(content);
 
     while (nbElements > 0) {
+        flag = 0;
+        // if the current page is not the first one (or last), a navigation bar exists
+        bool hasNav = !isLast || (pageIdxStart > 0) || (elemIdx > 0);
         if (content->type == TAG_VALUE_LIST) {
             nbElementsInPage = nbgl_useCaseGetNbTagValuesInPage(
                 nbElements, &content->content.tagValueList, elemIdx, &flag);
         }
+        else if (content->type == INFOS_LIST) {
+            nbElementsInPage = nbgl_useCaseGetNbInfosInPage(
+                nbElements, &content->content.infosList, elemIdx, hasNav);
+        }
+        else if (content->type == SWITCHES_LIST) {
+            nbElementsInPage = nbgl_useCaseGetNbSwitchesInPage(
+                nbElements, &content->content.switchesList, elemIdx, hasNav);
+        }
+        else if (content->type == BARS_LIST) {
+            nbElementsInPage = nbgl_useCaseGetNbBarsInPage(
+                nbElements, &content->content.barsList, elemIdx, hasNav);
+        }
+        else if (content->type == CHOICES_LIST) {
+            nbElementsInPage = nbgl_useCaseGetNbChoicesInPage(
+                nbElements, &content->content.choicesList, elemIdx, hasNav);
+        }
         else {
             nbElementsInPage = MIN(
                 nbMaxElementsPerContentType[content->type],
                 nbElements);  // TODO hardcoded to 3 for now but should be dynamically computed
```

Comment:
It is not modified by this PR, but is the comment still relevant?

Reply:
You're right, I'll remove it